source/driverHandler.py

```diff
@@ -0,0 +1,234 @@
+		self.displayName=displayName
+		self.availableInSettingsRing=availableInSettingsRing
+		self.useConfig = useConfig
+
+class NumericDriverSetting(DriverSetting):
+	"""Represents a numeric driver setting such as rate, volume, pitch or dot firmness."""
+
+	def _get_configSpec(self):
+		return "integer(default={defaultVal},min={minVal},max={maxVal})".format(
+			defaultVal=self.defaultVal,minVal=self.minVal,maxVal=self.maxVal)
+
+	def __init__(self, name, displayNameWithAccelerator, availableInSettingsRing=False,
+		defaultVal=50, minVal=0, maxVal=100, minStep=1, normalStep=5, largeStep=10,
+		displayName=None, useConfig=True):
+		"""
+		@param defaultVal: Specifies the default value for a numeric driver setting.
+		@type defaultVal: int
+		@param minVal: Specifies the minimum valid value for a numeric driver setting.
+		@type minVal: int
+		@param maxVal: Specifies the maximum valid value for a numeric driver setting.
+		@type maxVal: int
+		@param minStep: Specifies the minimum step between valid values for each numeric setting. For example, if L{minStep} is set to 10, setting values can only be multiples of 10; 10, 20, 30, etc.
+		@type minStep: int
+		@param normalStep: Specifies the step between values that a user will normally prefer. This is used in the settings ring.
+		@type normalStep: int
+		@param largeStep: Specifies the step between values if a large adjustment is desired. This is used for pageUp/pageDown on sliders in the Voice Settings dialog.
+		@type largeStep: int
+		@note: If necessary, the step values will be normalised so that L{minStep} <= L{normalStep} <= L{largeStep}.
+		"""
+		super(NumericDriverSetting,self).__init__(name, displayNameWithAccelerator, availableInSettingsRing=availableInSettingsRing,
+			displayName=displayName, useConfig=useConfig)
+		self.minVal=minVal
+		self.defaultVal=max(defaultVal,self.minVal)
+		self.maxVal=max(maxVal,self.defaultVal)
+		self.minStep=minStep
+		self.normalStep=max(normalStep,minStep)
+		self.largeStep=max(largeStep,self.normalStep)
+
+class BooleanDriverSetting(DriverSetting):
+	"""Represents a boolean driver setting such as rate boost or automatic time sync.
+	"""
+
+	def __init__(self, name, displayNameWithAccelerator, availableInSettingsRing=False,
+		displayName=None, defaultVal=False, useConfig=True):
+		"""
+		@param defaultVal: Specifies the default value for a boolean driver setting.
+		@type defaultVal: bool
+		"""
+		super(BooleanDriverSetting,self).__init__(name, displayNameWithAccelerator, availableInSettingsRing=availableInSettingsRing,
+			displayName=displayName, useConfig=useConfig)
+		self.defaultVal=defaultVal
+
+	def _get_configSpec(self):
+		return "boolean(default={defaultVal})".format(defaultVal=str(self.defaultVal).lower())
+
+class UnsupportedConfigParameterError(NotImplementedError):
+	"""
+	Raised when changing or retrieving a driver setting that is unsupported for the connected device.
+	"""
+
+class StringParameterInfo(object):

```

Comment:
Why doesn't `StringParameterInfo` also inherit from `DriverSetting`?

Reply:
Because it's not a setting, it is a parameter for a setting. There's no accelerator either, and I don't think parameter names are translatable.